### PR TITLE
add capacity on internal cluster via ui (selenium)

### PR DESCRIPTION
**Pull request by OdedViner**
Signed-off-by: Oded Viner <oviner@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/ui/infra_ui.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    """
+
+    def __init__(self, driver):
+        super().__init__(driver)
+        ocp_version = get_ocp_version()
+        self.infra_loc = locators[ocp_version]["infra"]
+        if config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM:
+            self.storage_class = "thin_sc"
+        elif config.ENV_DATA["platform"].lower() == constants.AWS_PLATFORM:
+            self.storage_class = "gp2_sc"
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/infra_ui.py:26]: same logic is already part of https://github.com/red-hat-storage/ocs-ci/blob/f120044486631f49133c9f3a137842673d765a1c/ocs_ci/deployment/deployment.py#L675

Can you please create a function out of these checks to be used by both places?

File: ocs_ci/ocs/ui/infra_ui.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class InfraUI(PageNavigator):
+    """
+    InfraUI Class
+
+    """
+
+    def __init__(self, driver):
+        super().__init__(driver)
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/infra_ui.py:20]: Could you please add explanation about the logic here of different classes?

File: tests/ui/test_add_capacity_ui.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        logging.info("Wait for osd pods to be in Running state")
+        for osd_pods in TimeoutSampler(
+            timeout=600,
+            sleep=10,
+            func=get_osd_pods,
+        ):
+            if len(osd_pods) == (osd_count + 3):
+                break
+
+        for osd_pod in osd_pods:
```
Comment (ebenahar) [inline tests/ui/test_add_capacity_ui.py:50]: is there validation we can do from the UI?

File: ocs_ci/ocs/ui/infra_ui.py (hunk 1 above)
Comment (OdedViner) [reply ocs_ci/ocs/ui/infra_ui.py:26]: yes, I will create `Common Function Class`
image

File: tests/ui/test_add_capacity_ui.py (hunk 3 above)
Comment (hnallurv) [reply tests/ui/test_add_capacity_ui.py:50]: Better to validate from UI. Newly added osd pods should be shown in UI too.

File: tests/ui/test_add_capacity_ui.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+class TestAddCapacityUI(object):
+    """
+    Test Add Capacity on via UI
+
+    """
+
+    @ui
+    @skipif_lso
+    @skipif_bm
+    @skipif_bmpsi
```
Comment (vasukulkarni) [inline tests/ui/test_add_capacity_ui.py:27]: instead of 2 above skips, you can use @cloud_platform_required (if it works only on aws, gcp ...)

**Comment by vasukulkarni** [comment]
LGTM

File: tests/ui/test_add_capacity_ui.py (hunk 4 above)
Comment (OdedViner) [reply tests/ui/test_add_capacity_ui.py:38]: it is work on vmware (internal cluster) too

File: ocs_ci/ocs/ui/infra_ui.py (hunk 2 above)
Comment (ebenahar) [reply ocs_ci/ocs/ui/infra_ui.py:20]: @OdedViner maybe add an explanation about this class in its docstring

File: ocs_ci/ocs/ui/infra_ui.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+from ocs_ci.ocs.ui.base_ui import PageNavigator
+from ocs_ci.ocs.ui.views import locators
+from ocs_ci.utility.utils import get_ocp_version
+
+
+logger = logging.getLogger(__name__)
+
+
+class InfraUI(PageNavigator):
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/infra_ui.py:11]: the class name does not reflect its purpose that is mentioned in the docstring below

**Review by vasukulkarni — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Pull request merged**